pkg/controller/podgc/gc_controller.go

```diff
@@ -124,6 +128,76 @@ func isPodTerminated(pod *v1.Pod) bool {
 	return false
 }
 
+// isPodTerminating returns true if the pod is terminating.
+func isPodTerminating(pod *v1.Pod) bool {
+	if phase := pod.Status.Phase; phase != v1.PodPending && phase != v1.PodRunning && phase != v1.PodUnknown {
+		return true
+	}
+	if pod.ObjectMeta.DeletionTimestamp != nil {
```

Comment:
I am thinking maybe  DeletionTimestamp is the only thing we should check here? Once node is not ready, we depends on eviction to mark eletiontimestamps on every pod?

Reply:
See lines 124-129, there is a `isPodTerminated`. I'm thinking `isPodTerminating` should include both `isPodTerminated` and if DeletionTimestamp is set.

Reply:
I think in this case, we only need to check DeletionTimestamp is set and then verify whether the node is out-of-service, for other pods in terminated state, the gc controller will take care.

Reply:
Sure.

Reply:
I will raise a commit to address this.

Reply:
Done